code/edgeconsent/src/main/java/com/adobe/marketing/mobile/edge/consent/ConsentExtension.java

```diff
@@ -35,8 +37,26 @@ class ConsentExtension extends Extension {
 	 * @param extensionApi {@link ExtensionApi} instance
 	 */
 	protected ConsentExtension(final ExtensionApi extensionApi) {
+		this(extensionApi, null);
+	}
+
+	/**
+	 * Constructor used for testing.
+	 * @param extensionApi {@link ExtensionApi} instance
+	 * @param namedCollection {@link NamedCollection} instance from {@link ServiceProvider}
+	 * @see ConsentExtension(ExtensionApi)
+	 */
+	protected ConsentExtension(final ExtensionApi extensionApi, final NamedCollection namedCollection) {
 		super(extensionApi);
-		consentManager = new ConsentManager();
+		consentManager =
+			new ConsentManager(
+				namedCollection == null
+					? ServiceProvider
+						.getInstance()
+						.getDataStoreService()
+						.getNamedCollection(ConsentConstants.DataStoreKey.DATASTORE_NAME)
+					: namedCollection
+			);
```

Comment:
Is it possible to move the generation of the NamedCollection here? I feel that's cleaner than checking for a null named collection in the overloaded constructor.
e.g.
```
protected ConsentExtension(final ExtensionApi extensionApi) {
		this(extensionApi, ServiceProvider
						.getInstance()
						.getDataStoreService()
						.getNamedCollection(ConsentConstants.DataStoreKey.DATASTORE_NAME));
	}
```

Reply:
@emdobrin will look at this and create a new PR if needed.

Reply:
Discussed with @emdobrin, updated.